craft_application/commands/lifecycle.py

```diff
@@ -357,8 +357,9 @@ def _fill_parser(self, parser: argparse.ArgumentParser) -> None:
 
         parser.add_argument(
             "--use-fetch-service",
-            action="store_true",
             help="Use the Fetch Service to inspect downloaded assets.",
+            choices=("strict", "permissive"),
```

Comment:
Minor nit: since this is now a set of choices, would it make more sense to set a `dest`? Maybe `dest="fetch_service_policy"`?